app/services/strategy_marketplace_service.py

```
@@ -1904,68 +2018,91 @@ async def _get_user_strategy_portfolio_impl(self, user_id: str) -> Dict[str, Any
-                            UserStrategyAccess.user_id == user_uuid,
-                            UserStrategyAccess.is_active.is_(True)
-                        )
+            user_uuid = uuid.UUID(user_id) if isinstance(user_id, str) else user_id
+        except ValueError:
+            self.logger.warning(
+                "Cannot load strategy access records - invalid user_id format",
+                user_id=user_id,
+            )
+            return [], None
+
+        async with get_database() as db:
```

Ruff (0.13.1)

Undefined name `get_database`

(F821)